snake_case FX IR names

ansley opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#50876 snake_case FX IR names**

Differential Revision: [D26002640](https://our.internmc.facebook.com/intern/diff/D26002640)

Commit:
snake_case FX IR names

[ghstack-poisoned]

jamesr66a commented on torch/fx/graph.py:
```diff
@@ -14,7 +14,23 @@ def _is_magic(x: str) -> bool:
+        res.append(s[i].lower())
+    if len(s) > 1:
+        res.append(s[-1].lower())
+    return ''.join(res)
```
I think using a loop-carried variable rather than doing +-1 indexing may be a bit cleaner:

```
    chars = []
    prev_upper = False
    for c in s:
        if prev_upper and c.islower():
            chars.append('_')
        chars.append(c.lower())

    return ''.join(chars)
```

jamesr66a replied:
I think using a loop-carried variable may be a bit cleaner than +-1 indexing:

```
def _snake_case(s: str) -> str:
    chars = []
    prev_lower = False
    for c in s:
        if prev_lower and c.isupper():
            chars.append('_')
        chars.append(c.lower())
        prev_lower = c.islower()

    return ''.join(chars)
```

jamesr66a reviewed (approved):
Nice!

Commit:
Update on "snake_case FX IR names"




[ghstack-poisoned]

Commit:
Update on "snake_case FX IR names"


Differential Revision: [D26002640](https://our.internmc.facebook.com/intern/diff/D26002640)

[ghstack-poisoned]

Commit:
Update on "snake_case FX IR names"


Differential Revision: [D26002640](https://our.internmc.facebook.com/intern/diff/D26002640)

[ghstack-poisoned]

Commit:
Update on "snake_case FX IR names"


Differential Revision: [D26002640](https://our.internmc.facebook.com/intern/diff/D26002640)

[ghstack-poisoned]
